### PR TITLE
Bump version containing default instance warmup feature

### PR DESCRIPTION
**Pull request by mnitchev**
### Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: helm/cluster-api-provider-aws/values.yaml [hunk 1]
```diff
@@ -3,7 +3,8 @@ name: cluster-api-provider-aws
 # needed. Please read https://github.com/giantswarm/cluster-api-provider-aws/blob/main/README.md on how to create a
 # release. Please include the short commit SHA in the tag name, such as `v2.0.2-gs-123abcd`. After changing this
 # tag, please run `make generate` to update CRDs and other manifests.
-tag: v2.3.0-gs-13a6101cc  # upstream v2.3.0 + backported features/fixes (https://github.com/giantswarm/cluster-api-provider-aws/pull/576) + more backports (https://github.com/giantswarm/cluster-api-provider-aws/pull/580 + https://github.com/giantswarm/cluster-api-provider-aws/pull/582 + https://github.com/giantswarm/cluster-api-provider-aws/pull/583)
+
```
Comment (AndiDog) [inline helm/cluster-api-provider-aws/values.yaml:6]: This should add a PR link

Comment (mnitchev) [reply helm/cluster-api-provider-aws/values.yaml:6]: Done, I also changed the format so it's a bit clearer

File: helm/cluster-api-provider-aws/values.yaml [hunk 2]
```diff
@@ -3,7 +3,13 @@ name: cluster-api-provider-aws
...
+# Changes in this tag include upstream v2.3.0 with additional:
+#   * backported features/fixes (https://github.com/giantswarm/cluster-api-provider-aws/pull/576)
+#   * more backports (https://github.com/giantswarm/cluster-api-provider-aws/pull/580)
+#   * https://github.com/giantswarm/cluster-api-provider-aws/pull/582
+#   * default instance warmup field addition (https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4760)
```
Comment (AndiDog) [inline helm/cluster-api-provider-aws/values.yaml:11]:
```suggestion
#   * https://github.com/giantswarm/cluster-api-provider-aws/pull/582
#   * https://github.com/giantswarm/cluster-api-provider-aws/pull/583
#   * default instance warmup field addition (https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4760)
```

**Review by AndiDog — COMMENTED** [review]
I'm adding back the accidentally-removed backport PR link

**Review by AndiDog — APPROVED** [review]

**Pull request merged**